app/src/features/apiClient/screens/apiClient/components/clientView/components/Collection/components/CollectionOverview/CollectionOverview.tsx

```diff
@@ -1,17 +1,85 @@
-import React from "react";
+import React, { useCallback, useState } from "react";
 import { RQAPI } from "features/apiClient/types";
+import { InlineInput } from "componentsV2/InlineInput/InlineInput";
+import { Input } from "antd";
+import { upsertApiRecord } from "backend/apiClient";
+import { useSelector } from "react-redux";
+import { getUserAuthDetails } from "store/slices/global/user/selectors";
+import { getCurrentlyActiveWorkspace } from "store/features/teams/selectors";
+import { useApiClientContext } from "features/apiClient/contexts";
+import { useDebounce } from "hooks/useDebounce";
+import { useTabsLayoutContext } from "layouts/TabsLayout";
+import PATHS from "config/constants/sub/paths";
 import "./collectionOverview.scss";
 
 interface CollectionOverviewProps {
   collection: RQAPI.CollectionRecord;
 }
 
 export const CollectionOverview: React.FC<CollectionOverviewProps> = ({ collection }) => {
+  const user = useSelector(getUserAuthDetails);
+  const team = useSelector(getCurrentlyActiveWorkspace);
+  const { onSaveRecord } = useApiClientContext();
+  const { replaceTab } = useTabsLayoutContext();
+
+  const [collectionName, setCollectionName] = useState(collection?.name || "");
+  const [collectionDescription, setCollectionDescription] = useState(collection?.description || "");
+
+  const handleDescriptionChange = useCallback(
+    async (value: string) => {
+      const newDescription = value;
+      const updatedCollection = {
+        ...collection,
+        description: newDescription,
+      };
+      return upsertApiRecord(user.details?.profile?.uid, updatedCollection, team?.id).then((result) => {
+        onSaveRecord(result.data);
+      });
+    },
+    [collection, onSaveRecord, user.details?.profile?.uid, team?.id]
+  );
+
+  const debouncedDescriptionChange = useDebounce(handleDescriptionChange, 1500);
+
+  const handleCollectionNameChange = async () => {
+    const updatedCollection = {
+      ...collection,
+      name: collectionName,
+    };
+    return upsertApiRecord(user.details?.profile?.uid, updatedCollection, team?.id).then((result) => {
+      onSaveRecord(result.data);
+      replaceTab(result.data.id, {
+        id: result.data.id,
+        title: result.data.name,
+        url: `${PATHS.API_CLIENT.ABSOLUTE}/collection/${result.data.id}`,
+      });
+    });
+  };
```

Comment:
based on how [variables were implemented](https://github.com/requestly/requestly/issues/2350) it seems like these should be moved inside [`CollectionView.tsx`](https://github.com/requestly/requestly/blob/master/app/src/features/apiClient/screens/apiClient/components/clientView/components/Collection/CollectionView.tsx#L67)

This is also how the auth tabs are being added to the collection view - https://github.com/requestly/requestly/pull/2411

Reply:
These functions should be with their respective component where they are being used which is the overview component. Moving these functions to CollectionView will add an unnecessary step to pass the functions as props to overview component.